test/unit/gateways/card_connect_test.rb

```diff
@@ -199,6 +199,17 @@ def test_scrub
     assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
   end
 
+  def test_frontendid_is_added_to_post_data_parameters
+    @gateway.class.application_id = 'my_app'
+    stub_comms(@gateway, :ssl_request) do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_, _, body|
+      assert_equal 'my_app', JSON.parse(body)['frontendid']
+    end.respond_with(successful_purchase_response)
+  ensure
+    @gateway.class.application_id = nil
+  end
```

Comment:
Do we need the `ensure` block? The `@gateway` instance variable and `application_id` should be reset for each test.

Reply:
I'm not sure. This test was adapted from https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/authorize_net_test.rb#L1011

Reply:
Verified this an existing pattern in place in ActiveMerchant.